src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -9,64 +8,108 @@ use std::time::Duration;
 use async_trait::async_trait;
 use bazel_protos::gen::build::bazel::remote::execution::v2 as remexec;
 use fs::RelativePath;
-use hashing::Digest;
+use hashing::{Digest, EMPTY_DIGEST};
 use maplit::hashset;
 use mock::{StubActionCache, StubCAS};
+use parking_lot::Mutex;
 use remexec::ActionResult;
 use store::{BackoffConfig, Store};
 use tempfile::TempDir;
 use testutil::data::{TestData, TestDirectory, TestTree};
-use testutil::relative_paths;
 use workunit_store::WorkunitStore;
 
 use crate::remote::{ensure_action_stored_locally, make_execute_request};
 use crate::{
   CommandRunner as CommandRunnerTrait, Context, FallibleProcessResultWithPlatform,
-  MultiPlatformProcess, NamedCaches, Platform, Process, ProcessMetadata,
+  MultiPlatformProcess, Platform, Process, ProcessMetadata,
 };
 
-struct RoundtripResults {
-  uncached: Result<FallibleProcessResultWithPlatform, String>,
-  maybe_cached: Result<FallibleProcessResultWithPlatform, String>,
+/// A mock of the local runner used for better hermeticity of the tests.
+#[derive(Clone)]
+struct MockLocalCommandRunner {
+  result: Result<FallibleProcessResultWithPlatform, String>,
+  call_counter: Arc<Mutex<u32>>,
```

Comment:
> More idiomatic is to use an atomic counter type like [AtomicUsize](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html).

I'm curious why you think it's more idiomatic to use an atomic type. My understanding is that the atomic types use some platform-specific overhead compared to the standard integer types, in order to provide thread safety, which we don't need in a test. So using an atomic type would add overhead for no particular gain here.

Reply:
Arc already has that same overhead give or take to implement its capital A IIUC. If threads aren't in play in tests the proof is dropping the Arc to an Rc and seeing if it compiles.

Reply:
> I'm curious why you think it's more idiomatic to use an atomic type. 

Just my impression that it is always nice to avoid a lock if possible, and atomic types let one do just that. This is test code so I'm not too concerned if this remains a `Mutex` for readability or other purposes.

Reply:
`Rc` breaks because we store this on `CommandRunner`, which must implement `Send`.

I'll go with `AtomicUsize` instead of `Mutex<u32>` for clearer semantics.